src/test/ui/parser/issue-68987-unmatch-issue.rs

```diff
@@ -0,0 +1,12 @@
+// This file has unexpected closing delimiter,
+
+fn func(o: Option<u32>) {
+    match o {
+        Some(_x) =>   // Missing '{'
+            let _ = if true {};
+        }
+        None => {}
+    }
```

Comment:
This case is making me think of how we should special case a few cases of malformed expressions like `match`, as we have quite a few flagposts to figure out what was intended.